entries/tests.py

```diff
@@ -1,13 +1,141 @@
+        self.item_2 = Item.objects.create(type="question", plio=self.plio_1, time=2)
+
+    def test_cannot_create_session_for_draft_plio(self):
+        response = self.client.post(reverse("sessions-list"), {"plio": self.plio_1.id})
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+        self.assertEqual(
+            json.loads(response.content)["plio"][0],
+            "A session can only be created for a published plio",
+        )
+
+    def test_can_create_session_for_published_plio(self):
+        # set plio_1 as published
+        self.plio_1.status = "published"
+        self.plio_1.save()
+        response = self.client.post(reverse("sessions-list"), {"plio": self.plio_1.id})
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+
+    def test_fresh_session_created_if_no_old_sessions_exist(self):
+        """
+        If no old sessions exist, a fresh session should be
+        created and should contain all default values
+        """
+        # publish plio_1
+        self.plio_1.status = "published"
+        self.plio_1.save()
+
+        # create a new session for `plio_1` by `user`
+        response = self.client.post(reverse("sessions-list"), {"plio": self.plio_1.id})
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+        self.assertTrue(response.data["is_first"])
+        self.assertEqual(len(response.data["session_answers"]), 2)
+        self.assertEqual(response.data["session_answers"][0]["item_id"], self.item_1.id)
+        self.assertEqual(response.data["session_answers"][1]["item_id"], self.item_2.id)
+        self.assertEqual(response.data["retention"], ("0," * self.video.duration)[:-1])
+
+    def test_old_session_details_used_if_old_sessions_exist(self):
+        """
+        If old sessions exist, a new session created should carry
+        over all the details from the older session
+        """
+        # publish plio_1
+        self.plio_1.status = "published"
+        self.plio_1.save()
+
+        # create a new, first session for plio_1
+        response = self.client.post(reverse("sessions-list"), {"plio": self.plio_1.id})
+        session_1 = response.data
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+        self.assertTrue(session_1["is_first"])
+
+        # update that session with some details
+        response = self.client.put(
+            reverse("sessions-detail", args=[session_1["id"]]),
+            {
+                "plio": self.plio_1.id,
+                "watch_time": 5.00,
+                "retention": "1,1,1,1,1,0,0,0,0,0",
+            },
+        )
+        session_1_updated = response.data
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertFalse(session_1_updated["is_first"])
+        self.assertEqual(session_1_updated["id"], session_1["id"])
+
+        # mocking reloading of the page, hence creating a new session
```

Comment:
should not be mentioning reload etc - that is how it is happening in the frontend - no need of mentioning it in the backend API.